### PR TITLE
IPDK indirect counter test case automation

### PR DESCRIPTION
**Pull request by sjwahub**
Signed-off-by: Shujun Wang <shujun.wang@intel.com>

DPDK  indirect counter test case automation

1.  Run test suite log. 
[ptf_tests_run13.log](https://github.com/ipdk-io/ovs/files/9523946/ptf_tests_run13.log)
where you would notice duplicated cases in summary. Actually they're different.  Currently ptf can't differentiate test cases based on json file. In this case, for example, dpdk_l3em_indirect_counter run with pna and psa json files separately.
2. rebuild ovs log 
[Indirect_CT_ovs_build.txt](https://github.com/ipdk-io/ovs/files/9523904/Indirect_CT_ovs_build.txt)

### CODE REVIEW

**Comment by ipdksw** [comment]
Result:FAILURE JenkinsConsole:https://s3.ipdk.io/ovs/PR-64-1/index.html GitCommit:https://github.com/ipdk-io/ovs/commits/ddc3097fc4e6f7f0c0e975736a65fea0753f0a7d

**Comment by ipdksw** [comment]
Result:FAILURE JenkinsConsole:https://s3.ipdk.io/ovs/PR-64-2/index.html GitCommit:https://github.com/ipdk-io/ovs/commits/30a292504aa701b2e16f4ff76dccf76f8f69fb75

**Comment by ipdksw** [comment]
Result:SUCCESS JenkinsConsole:https://s3.ipdk.io/ovs/PR-64-3/index.html GitCommit:https://github.com/ipdk-io/ovs/commits/1458a68efc7ec8bc9dd80d4481541c32f2188806

**Review by 5abeel — APPROVED** [review]
lgtm

**Comment by ipdksw** [comment]
Result:SUCCESS JenkinsConsole:https://s3.ipdk.io/ovs/PR-64-4/index.html GitCommit:https://github.com/ipdk-io/ovs/commits/1458a68efc7ec8bc9dd80d4481541c32f2188806

File: ptf_tests/common/utils/ovsp4ctl_utils.py [hunk 1]
```diff
@@ -284,4 +286,67 @@ def ovs_p4ctl_get_group(bridge, tbl_name, group_id, group_details=None):
...
+            
+        return counter
+    except Exception as error:
+        print(f"FAIL: ovs-p4ctl add group Failed with error: {error}")
+        return False
+        
+def ovs_p4ctl_get_counter_table_and_id():
+    """
+    :Function to check if netperf is installed and pkill running porcess
+    :and prepare to restart it.
```
Comment (dkaranjai) [inline ptf_tests/common/utils/ovsp4ctl_utils.py]: is this correct description? is it checking for netperf?

File: ptf_tests/common/utils/ovsp4ctl_utils.py [hunk 2]
```diff
@@ -284,4 +286,67 @@ def ovs_p4ctl_get_group(bridge, tbl_name, group_id, group_details=None):
     except Exception as error:
         print(f"FAIL: ovs-p4ctl get group Failed with error: {error}")
         return False
+    
+def ovs_p4ctl_get_counter_data(bridge, cnt_tbl_name, flow):
+    """
+    Args:
```
Comment (dkaranjai) [inline ptf_tests/common/utils/ovsp4ctl_utils.py]: please update the function description

File: ptf_tests/common/utils/test_utils.py [hunk 3]
```diff
@@ -639,3 +639,18 @@ def ip_ntns_exec_ping_test(nsname, dst_ip, count="4", remote=False, hostname="",
...
+
+def get_ovs_p4ctl_help(option):
+    """
+    :Function to get output of command "ovs-p4ctl --help"
+    :returns output or False
+    """
+    connection = Local()
+    cmd =f"ovs-p4ctl {option}"
+    output, _, _ = connection.execute_command(cmd)
+    connection.tear_down
```
Comment (dkaranjai) [inline ptf_tests/common/utils/test_utils.py]: tear_down()

**Review by dkaranjai — CHANGES_REQUESTED** [review]

**Comment by ipdksw** [comment]
Result:SUCCESS JenkinsConsole:https://s3.ipdk.io/ovs/PR-64-6/index.html GitCommit:https://github.com/ipdk-io/ovs/commits/298bf046a4c59ae6d6d5523564ddc345c9cfa66e

**Comment by ipdksw** [comment]
Result:SUCCESS JenkinsConsole:https://s3.ipdk.io/ovs/PR-64-7/index.html GitCommit:https://github.com/ipdk-io/ovs/commits/298bf046a4c59ae6d6d5523564ddc345c9cfa66e

**Comment by ipdksw** [comment]
Result:SUCCESS JenkinsConsole:https://s3.ipdk.io/ovs/PR-64-8/index.html GitCommit:https://github.com/ipdk-io/ovs/commits/a6eaf330fb12b0025f3038c1275fb34cef0a8abb

File: ptf_tests/common/utils/test_utils.py (hunk 3 above)
Comment (sjwahub) [reply ptf_tests/common/utils/test_utils.py]: fixed it.

**Comment by ipdksw** [comment]
Result:SUCCESS JenkinsConsole:https://s3.ipdk.io/ovs/PR-64-9/index.html GitCommit:https://github.com/ipdk-io/ovs/commits/8442b8482f3d019309c0de67a78fd145c71fdc12

**Comment by ipdksw** [comment]
Result:SUCCESS JenkinsConsole:https://s3.ipdk.io/ovs/PR-64-10/index.html GitCommit:https://github.com/ipdk-io/ovs/commits/8442b8482f3d019309c0de67a78fd145c71fdc12

**Review by dkaranjai — APPROVED** [review]

**Pull request merged**